pkg/spdk/spdk_test.go

```
@@ -111,7 +111,8 @@ func (s *TestSuite) TestSPDKBasic(c *C) {
 	c.Assert(err, IsNil)
 
 	// Do blindly cleanup
-	spdkCli.DeleteDevice(defaultDeviceName, defaultDeviceName)
+	err = spdkCli.DeleteDevice(defaultDeviceName, defaultDeviceName)
+	c.Assert(err, IsNil)
```

No need to validate the err because it does blindly cleanup.

---

If it's blindly handled, then the error should not be returned, correct? If yes, asserting here should be fine?

---

I meant below. However, we can still keep the returned err for the method, as it could be used later? or just remove the returned err. 

[/pkg/spdk/client/advanced.go#L41-L51](https://github.com/longhorn/go-spdk-helper/blob/220a127a613223825399c4b7717fd221b4cd8045/pkg/spdk/client/advanced.go#L41-L51)
```go
func (c *Client) DeleteDevice(bdevAioName, lvsName string) (err error) {
	if _, err := c.BdevLvolDeleteLvstore(lvsName, ""); err != nil {
        // log warning if needed
		return nil
	}

	if _, err := c.BdevAioDelete(bdevAioName); err != nil {
		 // log warning if needed
	    return nil
	}

	return nil
}
```

---

This means we always ignore any error. I think we can handle the returned error in the caller end rather than ignore it.

---

After discussing with @derekbit , one type of error is intended when the device is not found. When encountering this error, we can ignore it.

---

Or, we need to check if the device is present before deleting it.
It might better? WDYT?